src/RA_Dlg_Memory.cpp

```diff
@@ -782,661 +783,662 @@ void Dlg_Memory::ClearLogOutput() noexcept
-                            if (!m_SearchResults[m_nPage].m_results.GetMatchingAddress(nSelect - 2, result))
-                                break;
 
-                            ComboBox_SetText(GetDlgItem(hDlg, IDC_RA_WATCHING),
-                                             NativeStr(ra::ByteAddressToString(result.nAddress)).c_str());
+                        ComboBox_SetText(GetDlgItem(hwnd, IDC_RA_WATCHING),
+                                         NativeStr(ra::ByteAddressToString(result.nAddress)).c_str());
 
-                            const CodeNotes::CodeNoteObj* pSavedNote = m_CodeNotes.FindCodeNote(result.nAddress);
-                            if ((pSavedNote != nullptr) && (pSavedNote->Note().length() > 0))
-                                SetDlgItemTextW(hDlg, IDC_RA_MEMSAVENOTE, ra::Widen(pSavedNote->Note()).c_str());
-                            else
-                                SetDlgItemText(hDlg, IDC_RA_MEMSAVENOTE, _T(""));
-
-                            MemoryViewerControl::setAddress((result.nAddress & ~(0xf)) -
-                                                            ((int)(MemoryViewerControl::m_nDisplayedLines / 2) << 4) +
-                                                            (0x50));
-                            MemoryViewerControl::setWatchedAddress(result.nAddress);
-                            UpdateBits();
-                        }
+                        const CodeNotes::CodeNoteObj* pSavedNote = m_CodeNotes.FindCodeNote(result.nAddress);
+                        if ((pSavedNote != nullptr) && (pSavedNote->Note().length() > 0))
+                            SetDlgItemTextW(hwnd, IDC_RA_MEMSAVENOTE, ra::Widen(pSavedNote->Note()).c_str());
                         else
-                            ListView_SetItemState(GetDlgItem(hDlg, IDC_RA_MEM_LIST), -1, LVIF_STATE, LVIS_SELECTED);
+                            SetDlgItemText(hwnd, IDC_RA_MEMSAVENOTE, _T(""));
+
+                        MemoryViewerControl::setAddress(
+                            (result.nAddress & ~(0xf)) -
+                            (ra::to_signed(MemoryViewerControl::m_nDisplayedLines / 2) << 4) + (0x50));
+                        MemoryViewerControl::setWatchedAddress(result.nAddress);
+                        UpdateBits();
                     }
+                    else
+                        ListView_SetItemState(GetDlgItem(hwnd, IDC_RA_MEM_LIST), -1, LVIF_STATE, LVIS_SELECTED);
+
+                    return TRUE;
                 }
             }
         }
-        break;
+        return FALSE;
+    };
 
-        case WM_GETMINMAXINFO:
-        {
-            LPMINMAXINFO lpmmi = (LPMINMAXINFO)lParam;
-            lpmmi->ptMaxTrackSize.x = pDlgMemoryMin.x;
-            lpmmi->ptMinTrackSize = pDlgMemoryMin;
-        }
-            return TRUE;
+    const auto OnGetMinMaxInfo = []([[maybe_unused]] HWND, MINMAXINFO* restrict lpMinMaxInfo)
+    {
+        Expects(lpMinMaxInfo != nullptr);
```

Comment:
I put all these preconditions to make sure it was working in all configurations.